python/tank/descriptor/io_descriptor/base.py

```diff
@@ -693,6 +702,26 @@ def clone_cache(self, cache_root):
         filesystem.copy_folder(source_cache_path, new_cache_path, skip_list=[])
         return True
 
+    def attempt_move(self, src, dst):
+        """
+        Attempts to move a file/folder from source to destination.
+        :param src: The source file/folder.
+        :param dst: The destination file/folder.
+        """
+        # Proceed with the move only if the destination is not an existing folder.
+        # If the destination is a folder and exists, the move operation causes the source
+        # to be nested inside the destination folder, which is not the desirable behavior.
+        try:
+            if not os.path.isdir(dst):
+                shutil.move(src, dst)
+        except OSError as e:
+            # We choose to ignore any failed move attempts, as this might stem from
+            # concurrent processes moving a source to the same destination.
```

Comment:
You are leaving the download files on disk even tough they don't need to be moved. They should be deleted.